pennylane/plugins/default_qubit.py

```diff
@@ -388,78 +376,44 @@ def reset(self):
         self._state[0] = 1
 
-    def expand_one(self, U, wires):
-        r"""Expand a one-qubit operator into a full system operator.
+    def expand(self, U, wires):
+        r"""Expand a multi-qubit operator into a full system operator.
 
         Args:
-          U (array): :math:`2\times 2` matrix
-          wires (Sequence[int]): target subsystem
+          U (array): :math:`2^n \times 2^n` matrix where n = wires.
+          wires (Sequence[int]): Target subsystems (order matters!)
 
         Returns:
-          array: :math:`2^n\times 2^n` matrix
+          array: :math:`2^N\times 2^N` matrix. The full system operator.
         """
-        if U.shape != (2, 2):
-            raise ValueError('2x2 matrix required.')
-        if len(wires) != 1:
-            raise ValueError('One target subsystem required.')
-        wires = wires[0]
-        before = 2**wires
-        after = 2**(self.num_wires-wires-1)
-        U = np.kron(np.kron(np.eye(before), U), np.eye(after))
-        return U
-
-    def expand_two(self, U, wires):
-        r"""Expand a two-qubit operator into a full system operator.
-
-        Args:
-          U (array): :math:`4\times 4` matrix
-          wires (Sequence[int]): two target subsystems (order matters!)
+        if self.num_wires == 1:
+            # total number of wires is 1, simply return the matrix
+            return U
 
-        Returns:
-          array: :math:`2^n\times 2^n` matrix
-        """
-        if U.shape != (4, 4):
-            raise ValueError('4x4 matrix required.')
-        if len(wires) != 2:
-            raise ValueError('Two target subsystems required.')
+        N = self.num_wires
         wires = np.asarray(wires)
-        if np.any(wires < 0) or np.any(wires >= self.num_wires) or wires[0] == wires[1]:
+
+        if np.any(wires < 0) or np.any(wires >= N) or len(set(wires)) != len(wires):
             raise ValueError('Bad target subsystems.')
 
-        a = np.min(wires)
-        b = np.max(wires)
-        n_between = b-a-1  # number of qubits between a and b
-        # dimensions of the untouched subsystems
-        before = 2**a
-        after = 2**(self.num_wires-b-1)
-        between = 2**n_between
-
-        U = np.kron(U, np.eye(between))
-        # how U should be reordered
-        if wires[0] < wires[1]:
-            p = [0, 2, 1]
-        else:
-            p = [1, 2, 0]
-        dim = [2, 2, between]
-        p = np.array(p)
-        perm = np.r_[p, p+3]
-        # reshape U into another array which has one index per subsystem, permute dimensions, back into original-shape array
-        temp = np.prod(dim)
-        U = U.reshape(dim * 2).transpose(perm).reshape([temp, temp])
-        U = np.kron(np.kron(np.eye(before), U), np.eye(after))
-        return U
-
-    def expand_multi(self, U, wires):
-        r"""Expand a multi-qubit operator into a full system operator.
+        # generate N qubit basis states via the cartesian product
```

Comment:
use `itertools`, it'll likely be fastest

Reply:
Yep!
```python
In [17]: %timeit np.array(list(product([0, 1], repeat=4)))
9.87 µs ± 347 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

In [18]: %timeit np.stack(np.meshgrid(*[[0, 1] for _ in range(4)]), -1).reshape(-1, 4)
62.5 µs ± 1.38 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)
```

Reply:
lol also much clearer what it's doing :)